text/0000-api-version-compat.md

```diff
@@ -0,0 +1,121 @@
+
+If the API version declared in `CNB_PLATFORM_API` is higher than the newest supported by a given lifecycle, it will fail.
+
+### Buildpack API
+When orchestrating buildpacks, the lifecycle will provide inputs to and interpret outputs from each buildpack using the buildpack API indicated in that buildpack's `buildpack.toml` file.
+
+If the buildpack API declared in the buildpack.toml of a non-optional buildpack is higher than the newest supported by a given lifecycle, the lifecycle will fail.
+If the buildpack API declared in the buildpack.toml of an optional buildpack is higher than the newest supported by a given lifecycle, the lifecycle will warn, omit this buildpack from the group, and continue running.
+
+## Deprecations
+Once the 1.0 API is supported, we may wish to deprecate older APIs. A future RFC should provide a strategy for deprecations and outline the upgrade path from 0.x to 1.0 APIs. This is purposefully omitted from this RFC so that we can make progress on new 0.x APIs while we work out the details.
+
+# How it Works
+[how-it-works]: #how-it-works
+## Spec
+The core team must not allow changes in a 0.x API that necessarily prevents buildpacks implementing different APIs from running together.
+The contractual build plan is a danger zone for such changes. This does not necessarily mean no changes can be made to the contractual build plan,
+but any such changes should be made in close collaboration with the implementation team to ensure that the lifecycle can manage compatibility.
+
+## Lifecycle
+The lifecycle will maintain code for correctly implementing all supported APIs and choose which API to implement according to the rules above.
+
+## Platforms
+Platforms that wish to check lifecycle compatibility may now treat all minor API versions as non-breaking with the exception
+of buildpack and platform `0.1` -> `0.2`. Platforms should always set the `CNB_PLATFORM_API` environment variable to indicate
+which version of the API they are using.
+
+## Buildpacks
+Buildpacks should continue to declare the minimum required buildpack API in their `buildpack.toml`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The lifecycle must now deal with the complexity of implementing all 0.2+ APIs.
+
+The spec may not make certain types of changes to the contractual build plan before 1.0, even if those changes are desirable, because they make concurrent support for multiple APIs impossible.
+
+Buildpacks and platforms are likely to implement new APIs more slowly without a forcing function.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Keep doing what we are doing
+    - This will continue to cause pain for platform and buildpack authors
+- Do not allow breaking changes in the spec itself. Instead, add deprecations to the spec and print feature level deprecation warnings (the previous version of this RFC)
```

Comment:
Is there still a commit for this version? Is it possible to link to it?

Reply:
done